src/components/ExportButton/ExportButton.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from 'react';
+import { useRecoilValueLoadable } from 'recoil';
+import {
+  Box,
+  Fab,
+} from '@material-ui/core';
+import GetAppIcon from '@material-ui/icons/GetApp';
+
+import { routeDataQuery } from '../../state/route';
+import { LatLong } from '../../server/types';
+
+import {
+  convertCoordinatesToGPX,
+  downloadGPX,
+} from '../../state/utils';
+
+const ExportButton: React.FC = () => {
+  const route = useRecoilValueLoadable(routeDataQuery);
+  const [coordinates, setCoordinates] = useState<LatLong[]>();
+
+  const onExportButtonClick = (): void => {
+    if (!coordinates) return;
+
+    const gpx = convertCoordinatesToGPX(coordinates);
+    downloadGPX(gpx);
+  };
+
+  useEffect(() => {
+    if (route.state === 'hasValue' && route.contents) {
+      setCoordinates(route.contents.coordinates);
```

Comment:
If the data is available in `props` or through selectors I'd not copy the contents or to a local state.

This introduces possible out of sync data and creates unnecessary memory allocation and logic inside of the component. 

I've seen this pattern in class components in other projects, and it will come back to bite you in the ass later. So better not to use it like this.

Reply:
Good point! I'm not yet entirely familiar with the setup of this project, I expected some changes needing to be done ;).